packages/suite/src/components/suite/modals/TorLoading.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+
+import { desktopApi, BootstrapTorEvent } from '@trezor/suite-desktop-api';
+import { Button } from '@trezor/components';
+import { TorStatus } from '@suite-types';
+import { TorLoader, Modal, Translation } from '@suite-components';
+
+import type { UserContextPayload } from '@suite-actions/modalActions';
+
+type RequestEnableTorProps = Omit<
+    Extract<UserContextPayload, { type: 'request-enable-tor' }>,
+    'type'
+> & {
+    onCancel: () => void;
+};
+
+export const TorLoading = ({ onCancel, decision }: RequestEnableTorProps) => {
+    const [torStatus, setTorStatus] = useState<TorStatus>(TorStatus.Enabling);
+    const [progress, setProgress] = useState<number>(0);
+
+    useEffect(() => {
```

Comment:
What would you like to refactor? :)

Reply:
I like the way `useTor` hook is updating the state based on `tor/status`. 
So I would like to keep that way where the Tor status flow from main process to renderer comes throw that hook but extend it to use other kind of statuses.

So I am still thinking in different ways but one I have now in mind would be:

1. Move the logic that changes the Tor status based on `tor/bootstrap` to the Tor module in the main process.
2. Extend the hook `useTor` to update the state not only like Enable/Disable but Enabling/Disabling/Error and use `'tor/status'` to update it.
3. Then in this component just use the status that will be changing from the state
4. With this approach I am still not sure how would the right way to change the Tor status when user click on `Try again` or `Disable Tor`. In order to keep the `useTor` being the only way the state is being updated.

All of this does not work in the Tor splash screen because when that screen is loading the reducer is not loaded. So I am still thinking.

Reply:
You can extend `useTor` in `Main` _(setting the error status)_ without changing its role _(monitoring statuses sent from the Electron layer and saving to redux when needed)_ and have a `useTorStatus` hook that returns statuses, errors, and methods for user interactions. Basically `useTor` is a top-level singleton, `useTorStatus` is reusable :)